src/main/java/com/google/api/codegen/config/ConfigGeneratorApi.java

```diff
@@ -47,11 +49,30 @@
   public static final Option<String> OUTPUT_FILE =
       ToolOptions.createOption(
           String.class, "output_file", "The path of the output file to put generated config.", "");
+
+  private static final String CONFIG_KEY_TYPE = "type";
+  private static final String CONFIG_KEY_GENERATE_SAMPLES = "generate_samples";
+  private static final String CONFIG_KEY_LANGUAGE_SETTINGS = "language_settings";
+  private static final String CONFIG_KEY_INTERFACES = "interfaces";
+
   private static final String CONFIG_KEY_SERVICE_NAME = "name";
   private static final String CONFIG_KEY_METHOD_NAME = "name";
-  private static final String CONFIG_KEY_INTERFACES = "interfaces";
   private static final String CONFIG_KEY_METHODS = "methods";
   private static final String CONFIG_KEY_COLLECTIONS = "collections";
+  private static final String CONFIG_KEY_PACKAGE_NAME = "package_name";
+  private static final String CONFIG_KEY_REQUEST_OBJECT_METHOD = "request_object_method";
+
+  private static final String CONFIG_PROTO_TYPE = "com.google.api.codegen.ConfigProto";
+
+  private static final Map<String, String> LANGUAGE_PACKAGE_SEPARATORS =
+      ImmutableMap.<String, String>builder()
+          .put("java", ".")
+          .put("python", ".")
+          .put("go", ".")
+          .put("csharp", ".")
+          .put("ruby", "::")
+          .put("php", "\\")
+          .build();
```

Comment:
Node does not have anything in e.g. pubsub_gapic.yaml. It seems that Node packages are based on file location